src/components/context_menu/context_menu.tsx

```diff
@@ -47,6 +48,16 @@ export type EuiContextMenuPanelItemDescriptor = Omit<
   panel?: EuiContextMenuPanelId;
 };
 
+export interface EuiContextMenuPanelItemSeparator
+  extends EuiHorizontalRuleProps {
+  isSeparator: true;
+  key?: string;
+}
+
+export type EuiContextMenuPanelItemDescriptor =
+  | EuiContextMenuPanelItemDescriptorEntry
+  | EuiContextMenuPanelItemSeparator;
```

Comment:
```suggestion
export type EuiContextMenuPanelItemDescriptor = ExclusiveUnion<
  EuiContextMenuPanelItemDescriptorEntry,
  EuiContextMenuPanelItemSeparator
>;
```

where `ExclusiveUnion` can be imported from `'../common'`

TS correctly identifies the `isSeparator` prop as a discriminant, but does not flag additional properties as invalid if they come from the other side of the union. For example passing an item as `{ isSeparator: true, name: 'salad' },` is valid, as TS knows the `name` from the union's other type and that it won't be used by code because the discriminating `isSeparator` is present. But this fails on the developer experience side where it can appear `name` is still applicable and does something. So we introduced `ExclusiveUnion` to force TS to balk at entries that have no effect.

A fuller description with examples is included at https://github.com/elastic/eui/blob/cbb7f044fa11d8a8b244ce770fa3ad9aa7835d24/src/components/common.ts#L108

Reply:
Done.